<launch>
    <arg name="node_name" default="pid_tuning"/>
    <arg name="dynamic_reconfig_topic" default="dyn_config" />
    <arg name="kp" default="0.0" />
    <arg name="ki" default="0.0" />
    <arg name="kd" default="0.0" />


    <node name="$(arg node_name)"
          pkg="vesc_driver"
          type="dynamic_reconfig_vesc_driver"
          output="screen">
          <!-- ns="$(arg ns)"  -->

        <param name="kp_initial" value="$(arg kp)" />
        <param name="ki_initial" value="$(arg ki)" />
        <param name="kd_initial" value="$(arg kd)" />
        <param name="dynamic_reconfig_topic" value="$(arg dynamic_reconfig_topic)" />
        <remap from="/$(arg dynamic_reconfig_topic)/parameter_descriptions" to="$(arg dynamic_reconfig_topic)/parameter_descriptions"/>
        <remap from="/$(arg dynamic_reconfig_topic)/parameter_updates" to="$(arg dynamic_reconfig_topic)/parameter_updates"/>
        <remap from="/$(arg dynamic_reconfig_topic)/set_parameters" to="$(arg dynamic_reconfig_topic)/set_parameters"/>
        <remap from="$(arg node_name)/$(arg dynamic_reconfig_topic)" to="$(arg dynamic_reconfig_topic)" />
    </node>

</launch>